props/prop_base.h

```diff
@@ -435,12 +412,64 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #else
     SaberBase::SetVariation(0);
 #endif
+  }
 
+  // Select preset (font/style)
+  virtual void SetPreset(int preset_num, bool announce) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    FreeBladeStyles(preset_num);
```

Comment:
This is weird, because FreeBladeStyles() does current_preset_.SetPreset(), but then we do it again two lines down...

Reply:
FreeBladeStyles() only does UNSET_BLADE_STYLE